packages/web-components/src/components/rux-calendar/rux-calendar.tsx

```diff
@@ -0,0 +1,547 @@
+    */
+
+    //* This _date state controls the date that each compuatation uses. If the month updates, it'll update
+    //* this._date with that new month. Same for year.
+    //? Could we add a watch to this._date that will run the updateDate/State funcs?
+    @State() _date: Date = this.dateIn
+        ? utcToZonedTime(new Date(this.dateIn), 'UTC')
+        : utcToZonedTime(new Date(Date.now()), 'UTC')
+
+    @State() _month: number = this._date.getMonth() + 1 //getMonth returns a 0 indexed num, so we add 1
+    @State() _year: number = this._date.getFullYear()
+
+    @Watch('_month')
+    handleMonthWatch() {
+        console.log('heard month change')
+        this._updateDate(this._year, this._month)
+    }
+
+    @Watch('_year')
+    handleYearWatch() {
+        console.log('heard year change in WATCH')
+        // this._handleYears(this._maxDate, this._minDate)
+        this._updateDate(this._year, this._month)
+    }
+
+    //? Might need later when/if we tackle dynamic min/max changes
+    // @Watch('min')
+    // @Watch('max')
+    // handleMinMaxChange() {
+    //     console.log('heard min or max change')
+    // }
+
+    private _currentDate: Date = utcToZonedTime(new Date(Date.now()), 'UTC')
+    // private _year: number = this._date.getFullYear()
+    private _currentDay: number = this._currentDate.getDate()
+    private _daysInMonth: number = getDaysInMonth(this._date)
+    private _daysInMonthArr: Array<number> = []
+    private _prevMonth: number = this._month - 1 < 1 ? 12 : this._month - 1
+    private _nextMonth: number = this._month + 1 >= 13 ? 1 : this._month + 1
+    private _prevDaysToShow: { [key: string]: any } = {}
+    private _nextDaysToShow: Array<Number> = this._findNextDaysToShow()
+    private _allYearsArr: Array<Number> = []
+
+    //Default the max/min Dates to be 10 years in either direciton.
+    private _maxDate: Date = this.max
+        ? new Date(this.max)
+        : new Date(`${this._date.getFullYear() + 11}-01-01`)
+    private _minDate: Date = this.min
+        ? new Date(this.min)
+        : new Date(`${this._date.getFullYear() - 9}-01-01`)
+    private _maxYearArr: Array<Number> = []
+    private _minYearArr: Array<Number> = []
+
+    connectedCallback() {
+        console.log('CC Fire')
+
+        this._fillDaysInMonthArr()
+        this._nextDaysToShow = this._findNextDaysToShow()
+        this._handleMonthChange = this._handleMonthChange.bind(this)
+        this._handleYearChange = this._handleYearChange.bind(this)
+
+        if (this.dateIn) {
+            console.log('true')
+            console.log(this._maxDate, 'max date')
+            console.log(this._minDate, 'min date')
+            this._setStateWithDateIn()
+        }
+        this._handleYears(this._maxDate, this._minDate)
+    }
+    componentDidLoad() {
+        // attach event listeners to back and forward month arrows
+        this.el
```

Comment:
For getting the backward / forward elements, do you think it would be better to use the `ref` tagging pattern? That way you don't have to drill into the shadow, and potentially you could attach those listeners earlier.
This is the pattern I am referring to: setting a private variable `private _backArrow!: HTMLElement` and then setting the ref on the rux-icon in question to reference back to it: `ref={(el) => (this.backarrow = el!)}`. You should be able to use that as `this.backArrow.addEventListener('click', () => { my Func() })`

Reply:
I realized we don't even need to have a ref to the icons since they're being rendered always. I just threw the onClick onto them

Reply:
Even better!